src/class-parsely.php

```diff
@@ -818,6 +810,27 @@ public function display_admin_warning() {
 		<?php
 	}
 
+	/**
+	 * Decide wheter the admin display warning should be displayed
+	 *
+	 * @category Function
+	 * @package Parsely
+	 *
+	 * @return bool True if the admin warning should be displayed
+	 */
+	private function should_display_admin_warning() {
+		if ( is_network_admin() ) {
+			return false;
+		}
+
+		$options = $this->get_options();
+		if ( isset( $options['apikey'] ) && ! empty( $options['apikey'] ) ) {
```

Comment:
I think the `isset()` here can be dropped, as `empty()` already checks for that.

Then we're left with a negative condition that could be reversed and `return true;` instead, or simplify the whole condition down to `return empty( $options['apikey'] );`

I think the "get options and check if apikey is not empty populated" could be extracted to its own helper method, so `return ! $this->has_saved_api_key();` or `return $this->api_key_not_set();` or similar.

Reply:
I like the idea of simplification, but I think making a method just for `has_saved_api_key` might be a bit too much for the scope of this PR. I'm committing the first change you are suggesting.

Reply:
Opened https://github.com/Parsely/wp-parsely/issues/395 as checking for a saved API key is indeed a bigger issue.